server/impl/src/main/java/com/walmartlabs/concord/server/org/secret/SecretManager.java

```diff
@@ -351,13 +356,15 @@ public void update(String orgName, String secretName, SecretUpdateRequest req) {
 
         UUID finalProjectId = projectId;
 
+        UUID ownerId = req.ownerId();
```

Comment:
Other resources require `OWNER` privileges to change the owner. See `ProjectManager#update` for example.
I think we should follow the same logic here.

Reply:
I couldn't update the secret when I was signed in as another user while testing. could only do it when signed in as the owner

Reply:
That's the point.

Reply:
right. I meant we are already following the right logic, no?
what do I need to change then

Reply:
Interesting. Where's the check for that then? Similar to https://github.com/walmartlabs/concord/blob/master/server/impl/src/main/java/com/walmartlabs/concord/server/org/project/ProjectManager.java#L142-L151

The current code checks only for `WRITER` permissions https://github.com/walmartlabs/concord/blob/master/server/impl/src/main/java/com/walmartlabs/concord/server/org/secret/SecretManager.java#L265-L271

Reply:
there is no explicit check here because 

> An integration test would be nice :-)

yep will do

Reply:
ok so I added the check for secret#update#ownerId, but I still think the access level for that should be WRITER. 3 reasons - 
1. all other secret update options require only WRITER access, including delete method (which i changed to owner in this PR)
2. none of the secret#update actions, including ownerId change, can really harm anyone's concord setup. One can always recreate a secret as long as they have the password. if they don't, what are they doing as owners? :)
3. if the owner leaves the company, team members with write access should be able to change it without bothering us admins.
Secret#update is less critical than org and project in that sense. It's obviously your call in the end.. so let me know and i'll change it to OWNER if so

Reply:
@ibodrov

Reply:
I personally prefer symmetry with other kind of entities over the concerns above.

1. yes, the `DELETE` operation should require `OWNER` privileges;
2. sure, but that's where symmetry and consistency with other types of entities is better than a special case;
3. multiple people can be OWNERs including the org's OWNER.

Let's vote? @walmartlabs/concord :+1: this comment for requiring OWNER privileges, :+1: @d0s056f's post for lowering it to `WRITER`

Reply:
👍 for symmetry :)

Reply:
haha I'm convinced.. will do :)